engine-extensions/src/main/java/io/nosqlbench/engine/extensions/computefunctions/ComputeFunctions.java

```diff
@@ -66,14 +69,16 @@ public static double recall(long[] relevant, long[] actual) {
 
     public static double recall(long[] relevant, long[] actual, int k) {
         if (actual.length < k) {
-            throw new RuntimeException("indices fewer than limit, invalid precision computation: index count=" + actual.length + ", limit=" + k);
+            logger.warn("Returned indices fewer than limit in recall calculation: index count=" + actual.length + ", limit=" + k);
```

Comment:
Is this potentially an error condition which someone would want to gate on?
If so, we might want to promote it into an exception so that the error handler can handle it generically.
If not, then we might want to turn it into a metric (count the occurences).

Reply:
The actual condition is comparing lengths of actual and relevant, k is simply a parameter putting a ceiling on actual. Latest update addresses this and generates Exceptions as appropriate.